testproject/Assets/Scripts/SyncTransform.cs

```diff
@@ -38,7 +38,7 @@ public class SyncTransform : NetworkedBehaviour
 
         void SyncPosChanged(Vector3 before, Vector3 after)
         {
-            if (!IsLocalPlayer)
+            if (!IsOwner)
```

Comment:
@jeffreyrainy — we should use `IsOwner` rather than `IsLocalPlayer` here. we want owners to move transforms but not necessarily only players. non-player `NetworkObject`s should also be synchronized.